modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/web/DetectorController.java

```diff
@@ -134,4 +134,28 @@ public void updatedDetectorLastUsed(@RequestBody Map<String, String> params) {
     public void deleteDetector(@RequestParam String uuid) {
         service.deleteDetector(uuid);
     }
+
+    @GetMapping(path = "/getNextDetectorsToTrain", produces = "application/json")
+    @ResponseStatus(HttpStatus.OK)
+    public List<Detector> getNextDetectorsToTrain(@RequestHeader HttpHeaders headers) {
+        SpanContext parentSpanContext = trace.extractParentSpan(headers);
+        Span span = trace.startSpan("find-detectors-to-train-next", parentSpanContext);
+        val detectorList = service.getDetectorsToBeTrained();
+        span.finish();
+
+        return detectorList;
+    }
+
+    @PostMapping(path = "/updateDetectorTrainingTime")
```

Comment:
"updateDetectorTrainingNextRun" ?

Reply:
This method updates both values of the last train run time to current timestamp  & next training run to the value passed in the request param, that's why I kept its name general and not specific to updating next run